### PR TITLE
bugfix: fix access key loss after server restart

### LINKED ISSUE #5080: 客户端注册TM参数access-key问题
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
客户端配置access-key后注册TM会在服务端io.seata.server.auth.DefaultCheckAuthHandler#doRegTransactionManagerCheck接受到，但服务端重启后客户端再次注册到服务端时access-key为空

### Ⅱ. Describe what happened

服务端重启后，客户端再次注册则会出现此问题

### Ⅲ.  Environment:

- JDK version : 11
- Seata version:  1.5.2
- OS : windows 10

### PR DESCRIPTION
**Pull request by renliangyu857**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?
fixes #5080

### CODE REVIEW

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java [hunk 1]
```diff
@@ -137,55 +118,35 @@ public static TmNettyRemotingClient getInstance() {
...
+     * @return the instance
      */
-    protected void setSecretKey(String secretKey) {
-        if (null != secretKey) {
-            this.secretKey = secretKey;
-            return;
+    public static TmNettyRemotingClient getInstance() {
+        if (instance == null) {
+            synchronized (TmNettyRemotingClient.class) {
+                if (instance == null) {
```
Comment (a364176773) [inline core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java:136]: 如果为空应该直接抛异常提示先有带参的instance创建实例

**Review by a364176773 — APPROVED** [review]
LGTM @slievrly PTAL

File: core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java [hunk 2]
```diff
@@ -64,18 +62,26 @@ public final class TmNettyRemotingClient extends AbstractNettyRemotingClient {
...
+                                  ThreadPoolExecutor messageExecutor,
+                                  String applicationId,
+                                  String transactionServiceGroup,
+                                  String accessKey,
+                                  String secretKey) {
         super(nettyClientConfig, eventExecutorGroup, messageExecutor, NettyPoolKey.TransactionRole.TMROLE);
         this.signer = EnhancedServiceLoader.load(AuthSigner.class);
+        this.applicationId = applicationId;
+        this.transactionServiceGroup = transactionServiceGroup;
+        this.accessKey = accessKey;
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java:83]: It is recommended to check whether the value is null. Setting the value to null is meaningless. Use the setXXX call directly
---
这里建议判断是否为null，设置为null是无意义的。直接使用setXXX调用。

File: core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java [hunk 3]
```diff
@@ -91,17 +97,6 @@ public void onChangeEvent(ConfigurationChangeEvent event) {
...
 
-    /**
-     * Gets instance.
-     *
-     * @param applicationId           the application id
-     * @param transactionServiceGroup the transaction service group
-     * @return the instance
-     */
-    public static TmNettyRemotingClient getInstance(String applicationId, String transactionServiceGroup) {
-        return getInstance(applicationId, transactionServiceGroup, null, null);
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java:102]: It is recommended to retain the method without the ak/sk parameter.

---

建议保留无ak/sk 参数的方法。

File: core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java [hunk 4]
```diff
@@ -137,55 +118,23 @@ public static TmNettyRemotingClient getInstance() {
...
-     * Sets access key.
-     *
-     * @param accessKey the access key
-     */
-    protected void setAccessKey(String accessKey) {
-        if (null != accessKey) {
-            this.accessKey = accessKey;
-            return;
-        }
-        this.accessKey = System.getProperty(SEATA_ACCESS_KEY);
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/TmNettyRemotingClient.java:175]: It is recommended to keep this part, but also to determine if it is null.

---

建议保留，但同样需要判空。

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**